coq/coq-system.el

```diff
@@ -22,6 +22,15 @@
   (defvar coq-prog-args)
   (defvar coq-debug))
 
+;; Arbitrary arguments can already be passed through _CoqProject.
+;; However this is not true for all assistants, so we don't modify the
+;; (defpgcustom prog-args) declaration.
+(defun coq--string-list-p (obj)
+  "Determine if OBJ is a list of strings."
+  (or (null obj) (and (consp obj) (stringp (car obj)) (coq--string-list-p (cdr obj)))))
+
+(put 'coq-prog-args 'safe-local-variable #'coq--string-list-p)
```

Comment:
Is it safe to do this here (are we sure that coq-prog-args is defined at this point?)

Reply:
>Is it safe to do this here (are we sure that coq-prog-args is defined at this point?)

There's already `(eval-when-compile (defvar coq-prog-args))` above because it's used in some functions in this file.

Also I don't think it matters whether it's already defined.

Reply:
> There's already (eval-when-compile (defvar coq-prog-args)) above because it's used in some functions in this file.

That declares the variable, but it doesn't define it.

> Also I don't think it matters whether it's already defined.

Indeed, good point :) Thanks!